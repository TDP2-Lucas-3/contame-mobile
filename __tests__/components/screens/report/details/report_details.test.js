import React from 'react';
import {shallow} from 'enzyme';
import ReportDetails from '../../../../../src/components/screens/reports/details/report_details';

describe('ReportDetails', () => {
  let render;
  let props;

  beforeEach(() => {
    props = {
      report: {
        title: 'a report',
        description: 'some description',
        images: ['some image data'],
        location: {
          latitude: -1,
          longitude: -1,
        },
        creationDate: new Date('11/18/2020'),
        category: {},
      },
    };
    render = (appProps) => shallow(<ReportDetails {...appProps} />);
  });

  it('renders without crashing', () => {
    expect(() => render(props)).not.toThrowError();
  });

  it('renders all values correctly', () => {
    expect(render(props)).toMatchSnapshot();
  });
});
