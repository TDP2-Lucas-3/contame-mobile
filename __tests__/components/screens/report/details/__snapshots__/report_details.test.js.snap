// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportDetails renders all values correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F4D35E",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "zIndex": 9,
      }
    }
  >
    <SliderBox
      ImageComponent={[Function]}
      images={
        Array [
          "some image data",
        ]
      }
      style={
        Object {
          "backgroundColor": "#bdc6cf",
          "height": 200,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#1F271B",
          "borderBottomRightRadius": 10,
          "borderTopRightRadius": 10,
          "padding": 15,
          "position": "absolute",
          "top": "82%",
          "width": "85%",
        }
      }
    >
      <Themed.Text
        h4={true}
        style={
          Array [
            Object {
              "marginLeft": 20,
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        A report
      </Themed.Text>
    </View>
    <StatusBadge
      styles={
        Object {
          "margin": 15,
          "padding": 5,
          "position": "absolute",
          "right": 0,
          "top": -200,
        }
      }
      textStyle={
        Object {
          "fontSize": 14,
        }
      }
    />
  </View>
  <ScrollView>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginTop": 90,
          },
          Object {
            "margin": 30,
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Themed.Icon
        containerStyle={
          Object {
            "backgroundColor": "white",
            "borderRadius": 15,
            "elevation": 10,
            "height": 40,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
            "width": 40,
          }
        }
        name="calendar"
        type="font-awesome-5"
      />
      <Themed.Text
        style={
          Array [
            Object {
              "marginLeft": 20,
            },
            Object {
              "color": "#1F271B",
            },
          ]
        }
      >
        18 de November, 2020
      </Themed.Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginLeft": 30,
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Themed.Icon
        containerStyle={
          Object {
            "backgroundColor": "white",
            "borderRadius": 15,
            "elevation": 10,
            "height": 40,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
            "width": 40,
          }
        }
        name="map-marker"
        type="font-awesome-5"
      />
      <Themed.Text
        style={
          Array [
            Object {
              "marginLeft": 20,
            },
            Object {
              "color": "#1F271B",
            },
          ]
        }
      >
        <Component />
      </Themed.Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginLeft": 30,
          },
          Object {
            "marginTop": 30,
          },
          Object {
            "alignItems": "center",
          },
        ]
      }
    >
      <Themed.Icon
        containerStyle={
          Object {
            "backgroundColor": "white",
            "borderRadius": 15,
            "elevation": 10,
            "height": 40,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 5,
              "width": 0,
            },
            "shadowOpacity": 0.34,
            "shadowRadius": 6.27,
            "width": 40,
          }
        }
        name="layer-group"
        type="font-awesome-5"
      />
      <Themed.Text
        style={
          Array [
            Object {
              "marginLeft": 20,
            },
            Object {
              "color": "#1F271B",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 30,
          },
          Object {
            "marginTop": 20,
          },
          Object {
            "marginRight": 20,
          },
        ]
      }
    >
      <Themed.Text
        h4={true}
        style={
          Object {
            "color": "#1F271B",
          }
        }
      >
        Detalles
      </Themed.Text>
      <Themed.Text
        style={
          Array [
            Object {
              "marginTop": 20,
            },
            Object {
              "color": "#1F271B",
            },
          ]
        }
      >
        some description
      </Themed.Text>
    </View>
  </ScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#1F271B",
        "borderTopColor": "#86939e",
        "borderTopWidth": 1,
        "padding": 15,
      }
    }
  >
    <VoteButton />
  </View>
</View>
`;
